diff --git a/dependencies/dawn/CMakeLists.txt b/dependencies/dawn/CMakeLists.txt
index 9a90f5b74..fb8510249 100644
--- a/dependencies/dawn/CMakeLists.txt
+++ b/dependencies/dawn/CMakeLists.txt
@@ -50,10 +50,10 @@ set(CMAKE_DEBUG_POSTFIX "")
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 set(IS_DEBUG_BUILD 0)
-string(TOUPPER "${CMAKE_BUILD_TYPE}" build_type)
-if ((NOT ${build_type} STREQUAL "RELEASE") AND (NOT ${build_type} STREQUAL "RELWITHDEBINFO"))
-  set(IS_DEBUG_BUILD 1)
-endif()
+# string(TOUPPER "${CMAKE_BUILD_TYPE}" build_type)
+# if ((NOT ${build_type} STREQUAL "RELEASE") AND (NOT ${build_type} STREQUAL "RELWITHDEBINFO"))
+#   set(IS_DEBUG_BUILD 1)
+# endif()
 
 include(DawnSetIfNotDefined)
 
diff --git a/dependencies/dawn/third_party/CMakeLists.txt b/dependencies/dawn/third_party/CMakeLists.txt
index 8ff1eb05f..dadbbebda 100644
--- a/dependencies/dawn/third_party/CMakeLists.txt
+++ b/dependencies/dawn/third_party/CMakeLists.txt
@@ -117,7 +117,7 @@ endif()
 # End of Emscripten enabled third party directories
 ################################################################################
 if (EMSCRIPTEN)
-    return()
+    # return()
 endif()
 
 # Force ENABLE_RTTI in spirv-tools and glslang.
